### PR TITLE
doc: first draft of a Dafny ref manual in git markdown

### PR DESCRIPTION
**Pull request by davidcok**
-- needs title, TOC, bibliography, inline math and equations, section numbering, css, etc.

### CODE REVIEW

**Comment by richardlford** [comment]
Is this intended to replace Madoko?

**Comment by davidcok** [comment]
That is the plan. The principal drawback is support for math notation. The concern is for long-term support for madoko. Comments welcome. Sop far it is a few-hour experiment.
- David

File: Docs/DafnyRefMd/DafnyRef.md [hunk 1]
```diff
@@ -0,0 +1,8581 @@
...
+
+The main part of the reference manual is in top down order except for an
+initial section that deals with the lowest level constructs.
+
+[Co-induction Simply]: http://research.microsoft.com/en-us/um/people/leino/papers/krml230.pdf  "Co-induction Simply: Automatic Co-inductive Proofs in a Program Verifier"
+
+## Dafny Example
+To give a flavor of Dafny, here is the solution to a competition problem.
+
+```
```
Comment (robin-aws) [inline Docs/DafnyRefMd/DafnyRef.md:208]: If use "```dafny" on all the code snippets we'll get syntax highlighting! :)

File: Docs/DafnyRefMd/Makefile [hunk 2]
```diff
@@ -0,0 +1,3 @@
+all:
+	pandoc -t pdf -o DafnyRef.pdf --pdf-engine=xelatex --toc DafnyRef.md
```
Comment (robin-aws) [inline Docs/DafnyRefMd/Makefile:2]: We should get #605 merged in and use it here - I created that precisely to get the syntax highlighting to work for pandoc.

**Comment by davidcok** [comment]
This PR is obsolete.

**Pull request closed without merging**